src/middleware.js

```diff
@@ -65,23 +70,47 @@ export default function ({ onUnauthorized, ...options }) {
       mode,
     })
 
+    // Make the request
     return http(url, requestOptions)
       .catch(error => {
         const response = error.response || error.message || 'There was an error.'
-        next({
-          type: errorType,
-          payload: { ...rest, response },
-          error: true,
-        })
+        
+        // Send error action
+        if (errorAction) {
+          next(parseAction({
+            action: errorAction,
+            payload: { ...rest, response },
+            error: true,
+          }))
+        }
+
         if (error.status === 401 && defaultConfigOptions.onUnauthorized) {
           next(defaultConfigOptions.onUnauthorized())
         }
       })
-      .then(response =>
-        next({
-          type: successType,
-          payload: { ...rest, response },
-        })
-      )
+      .then(response => {
+
+        // Send success action
+        if (successAction) {
+          next(parseAction({
+            action: successAction,
+            payload: { ...rest, response },
+          }))
+        }
+        
+      })
   }
 }
+
+// Create an action from an action "definition."
+const parseAction = ({ action, payload={}, error=false }) => {
```

Comment:
hmm, just tested in node repl and seems fine (so much nicer now that these features are actually in node...) I think my issue was due to either removing keys for undefined values (that I am now handling with `omitUndefined`) or needing an explicit `false` value or something. Whatevs, looks good, my bad.